packages/office-ui-fabric-react/src/components/Shimmer/ShimmerElementsGroup/ShimmerElementsGroup.base.tsx

```diff
@@ -15,138 +15,127 @@ const getClassNames = classNamesFunction<IShimmerElementsGroupStyleProps, IShimm
 /**
  * {@docCategory Shimmer}
  */
-export class ShimmerElementsGroupBase extends BaseComponent<IShimmerElementsGroupProps, {}> {
-  public static defaultProps: IShimmerElementsGroupProps = {
-    flexWrap: false
-  };
-
-  private _classNames: { [key in keyof IShimmerElementsGroupStyles]: string };
-
-  constructor(props: IShimmerElementsGroupProps) {
-    super(props);
-  }
+export const ShimmerElementsGroupBase: React.FunctionComponent<IShimmerElementsGroupProps> = props => {
+  const { styles, width, shimmerElements, rowHeight, flexWrap = false, theme, backgroundColor } = props;
+
+  const classNames = getClassNames(styles!, {
+    theme: theme!,
+    flexWrap
+  });
+
+  const height = rowHeight ? rowHeight : findMaxElementHeight(shimmerElements ? shimmerElements : []);
+
+  return (
+    <div style={{ width: width ? width : 'auto' }} className={classNames.root}>
+      {getRenderedElements(shimmerElements, backgroundColor, height)}
+    </div>
+  );
+};
+
+function getRenderedElements(shimmerElements?: IShimmerElement[], backgroundColor?: string, rowHeight?: number): React.ReactNode {
+  const renderedElements: React.ReactNode = shimmerElements ? (
+    shimmerElements.map(
+      (element: IShimmerElement, index: number): JSX.Element => {
+        const { type, ...filteredElem } = element;
+        switch (element.type) {
+          case ShimmerElementType.circle:
+            return <ShimmerCircle key={index} {...filteredElem} styles={getElementStyles(element, backgroundColor, rowHeight)} />;
+          case ShimmerElementType.gap:
+            return <ShimmerGap key={index} {...filteredElem} styles={getElementStyles(element, backgroundColor, rowHeight)} />;
+          case ShimmerElementType.line:
+            return <ShimmerLine key={index} {...filteredElem} styles={getElementStyles(element, backgroundColor, rowHeight)} />;
+        }
+      }
+    )
+  ) : (
+    <ShimmerLine height={ShimmerElementsDefaultHeights.line} styles={{ root: [{ borderWidth: '0px' }] }} />
```

Comment:
Nit, since the above line of `{ root: [{ borderWidth: '0px' }] }` is not dynamic, it might make sense to hoist and reduce the need for invoking mergeStyles per render.